ts/screens/services/ServicesHomeScreen.tsx

```diff
@@ -521,7 +521,13 @@ class ServicesHomeScreen extends React.Component<Props, State> {
         tabContainerStyle={[styles.tabBarContainer, styles.tabBarUnderline]}
         tabBarUnderlineStyle={styles.tabBarUnderlineActive}
         onChangeTab={(evt: any) => {
-          this.setState({ currentTab: evt.i, isLongPressEnabled: false });
+          const currentTab: number = evt.i;
+          this.setState({
```

Comment:
If you are setting a state value by reading a state value you could have an inconsistent state update

> setState is not guaranteed to update state immediately, state updates are batched

So a better approach could be:

```typescript
const currentTab: number = evt.i;
const sameTab = this.state.currentTab === currentTab;
const isLongPressEnabled = this.state.isLongPressEnabled;
this.setState({
	currentTab,
	isLongPressEnabled: sameTab && isLongPressEnabled
});
```

Reply:
what about this approach?

```ts
const { currentTab, isLongPressEnabled } = this.state;
const nextTab: number = evt.i;
const isSameTab = currentTab === nextTab;
this.setState({
            currentTab: nextTab,
            isLongPressEnabled: isSameTab && isLongPressEnabled
});
```